clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -265,7 +265,12 @@ object UncommittedGarbageCollector {
     if (!markedRunSummary.first.getAs[Boolean]("success")) {
       throw new FailedRunException(s"Provided mark ($markID) is of a failed run")
     } else {
-      spark.read.parquet(s"$reportPath/deleted")
+      val deletedPath = new Path(formatRunPath(storageNamespace, markID) + "/deleted")
+      if (!fs.exists(deletedPath)) {
+        println(s"Mark ID ($markID) does not contain deleted files")
+        spark.emptyDataFrame.withColumn("address", lit(""))
```

Comment:
This empty dataframe will be created and dropped.  I think an "else" may be missing on the next line?

Reply:
Good catch! 👍
This works since we always create a deleted parquet even if no addresses to delete. I'll open another PR